docs/Advanced-Usage/FireAxe-Partitioning-onto-Multiple-FPGAs/FireAxe-Overview.rst

```diff
@@ -0,0 +1,73 @@
+enabled researchers to generate new SoC configurations that no longer fit in a
+single FPGA. FireAxe builds on top of FireSim to enable *partitioning of large
+designs onto multiple FPGAs*, overcoming the single-FPGA limitation for monolithic RTL designs.
+
+Partition Modes
+==================
+FireSim uses host-decoupling to control the advancement of the simulator, and
+this causes weird things to happen when the design is partitioned across multiple FPGAs.
+FireAxe provides users with three options (modes) to perform partitioning : exact-mode, fast-mode and the NoC-partition-mode.
+These options are passed on to the compiler which will generate the correct circuitry
+to deal with the odd things that happen on the partition interface.
+A more detailed explanation about the circuitry and the partition modes are provided in
+`the FireAxe paper <https://joonho3020.github.io/assets/ISCA2024-FireAxe.pdf>`_.
+However, if you just want to use FireAxe, the below explanations are sufficient
+for you to get started.
+
+Exact-Mode
+-----------
+In the exact-mode, users can choose the modules to partition out and place on separate FPGAs.
+The partitioned simulation will behave *exactly* the same as when running the target on a software RTL simulator.
+This is usesful when the partition boundary is not latency insensitive (i.e. the interface is not ready-valid interface nor credit based)
+and when the boundary contains combinational logic running through it.
+
+Example modules that can be partitioned out are RoCC accelerators as the ports to access the
+page-table-walker boundaries are combinationally dependent to each other.
+However, when running baremetal tests, address translations are unnecessary
+and users can use the fast-mode to partition out RoCC accelerators.
+
+Fast-Mode
+----------
+When the partition boundary is latency insensitive (i.e., the interface is ready-valid or credit based),
+you can use the fast-mode to perform partitioned simulations.
+Similarily as in the exact-mode, users can choose the modules to partition out.
+However, in the fast-mode, it provides higher simulation throughput by trading
+off simulation accuracy with performance. By injecting a single cycle of latency
+on the partition boundary, simulated design will run nearly 2x faster than the exact-mode.
+
+Example modules that can be partitioned out are core tiles as the tile to bus
+connections are ready-valid and the interrupt signals are also latency insensitive.
+In practice, adding a cycle of latency on the partition boundary has nearly zero accuracy implications.
+
+NoC-Partition-Mode
+------------------
+In the NoC partition mode, we exploit the fact that the NoC router boundaries are latency insensitive (credit based).
+Users simply has to specify the router nodes to be grouped together and the compiler will automatically group the modules
+to partition out with the selected routers. The NoC-partition-mode works only when partitioning tiles out.
```

Comment:
Why is only tiles a limitation? Why can't you export out say the nodes of the cache?

Reply:
The cache is difficult to partition out as it is connected to the `mbus` which complicates things.
Tiles are a reasonable target as it is the most resource consuming part of the design, and it is "the" module that you increase when you build scale up chips.
Also, the bounds the complexity of the NoC partitioning pass which is already quite complicated.